icon/common.js

```diff
@@ -0,0 +1,70 @@
+define([
+	"../has"
+], function (has) {
+
+	var SPRITE_CONTAINER_ID = 'requirejs-dplugins-icons';
+
+	return {
+		cache: {}, // stores all required paths
+		icons: [], // stores markup of all fetched icons
+		spriteContainer: null, // DOM element, parent of svg sprite
+		getFilename: getFilename,
+		createContainer: createContainer,
+		extractIcon: extractIcon,
+		generateSprite: generateSprite
+	};
+
+	// takes a path and returns the filename
+	function getFilename(filepath) {
+		return filepath.replace(/.*\/(.*)\.svg$/g, "$1");
+	}
+
+	// Creates the sprite container. This method should be executed once only.
+	function createContainer() {
+		var div = document.createElement("div");
+		div.setAttribute("id", SPRITE_CONTAINER_ID);
+		document.body.appendChild(div);
+		return div;
+	}
+
+	// Makes a symbol out of an svg icon
+	function extractIcon(svg, filename, cheerio /* SVG parser */) {
+		var icon, id, viewBox;
+		if (has("builder") && cheerio) { // server side SVG parsing
```

Comment:
the key difference is that client side, I use the browser's parser. 
I thought making the distinction server vs. client made more sense to someone who reads this function.